mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java

```diff
@@ -272,7 +275,8 @@ public int insert(String table, String key,
                 BulkWriteResult result = collection.withWriteConcern(
                         writeConcern).bulkWrite(bulkInserts,
                         new BulkWriteOptions().ordered(false));
-                if (result.getInsertedCount() == bulkInserts.size()) {
+                if (!result.wasAcknowledged()
```

Comment:
The short answer is yes.

A few things to make sure we are on the same page.
* 0 is "success"
* the "wasAcknowledged()" returns true if the user requested an 
acknowledgment from the server via the WriteConcern.

So this is saying return success if the user did _not_ request an 
acknowledgment from the server.

The user of YCSB controls if an acknowledgment is requested via the 
MongoDB URL's WriteConcern.  w=0 in the URL (e.g., 
mongodb://locahost:27017/?w=0) mean no acknowledgment. Anything else 
requests an acknowledgment, including w=1 which is the default.

For completeness, the second half of the || handles the case where there 
is an requested acknowledgment by checking the inserted count.

Rob.


On 06/07/2015 01:45 PM, Connor McCoy wrote:
> In mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java
> <https://github.com/brianfrankcooper/YCSB/pull/285#discussion_r31878835>:
>
>> @@ -272,7 +275,8 @@ public int insert(String table, String key,
>>                  BulkWriteResult result = collection.withWriteConcern(
>>                          writeConcern).bulkWrite(bulkInserts,
>>                          new BulkWriteOptions().ordered(false));
>> -                if (result.getInsertedCount() == bulkInserts.size()) {
>> +                if (!result.wasAcknowledged()
>
> Is this correct? It looks like you'll return 0 if the result was /not/
> acknowledged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brianfrankcooper/YCSB/pull/285/files#r31878835>.
>


-- 
Robert J. Moore
President
Allanbank Consulting, Inc.
Robert.J.Moore@allanbank.com